src/data/onChainPoolDataViaReadContract.ts

```diff
@@ -0,0 +1,451 @@
+                    parse: (results: Results, shift: number) => ({
+                        ...defaultCallsAux.parse(results, shift),
+                        ...weightedCalls.parse(
+                            results,
+                            shift + defaultCallsAux.count,
+                        ),
+                    }),
+                };
+        }
+        case 'Stable': {
+            if (poolTypeVersion === 1) {
+                return {
+                    count: defaultCalls.count + stableCalls.count,
+                    build: (id: string) => [
+                        ...defaultCalls.build(id, poolType),
+                        ...stableCalls.build(id),
+                    ],
+                    parse: (results: Results, shift: number) => ({
+                        ...defaultCalls.parse(results, shift),
+                        ...stableCalls.parse(
+                            results,
+                            shift + defaultCalls.count,
+                        ),
+                    }),
+                };
+            } else {
+                return {
+                    count: defaultCallsAux.count + stableCalls.count,
+                    build: (id: string) => [
+                        ...defaultCallsAux.build(id, poolType),
+                        ...stableCalls.build(id),
+                    ],
+                    parse: (results: Results, shift: number) => ({
+                        ...defaultCallsAux.parse(results, shift),
+                        ...stableCalls.parse(
+                            results,
+                            shift + defaultCallsAux.count,
+                        ),
+                    }),
+                };
+            }
+        }
+        case 'StablePhantom':
+        case 'MetaStable':
+            return {
+                count: defaultCalls.count + stableCalls.count,
+                build: (id: string) => [
+                    ...defaultCalls.build(id, poolType),
+                    ...stableCalls.build(id),
+                ],
+                parse: (results: Results, shift: number) => ({
+                    ...defaultCalls.parse(results, shift),
+                    ...stableCalls.parse(results, shift + defaultCalls.count),
+                }),
+            };
+        case 'ComposableStable': {
+            return {
+                count: defaultCallsAux.count + stableCalls.count,
+                build: (id: string) => [
+                    ...defaultCallsAux.build(id, poolType),
+                    ...stableCalls.build(id),
+                ],
+                parse: (results: Results, shift: number) => ({
+                    ...defaultCallsAux.parse(results, shift),
+                    ...stableCalls.parse(
+                        results,
+                        shift + defaultCallsAux.count,
+                    ),
+                }),
+            };
+        }
+        case 'GyroE':
+            if (poolTypeVersion === 2) {
```

Comment:
Why reverse the logic here? Doesn't it make more sense to assume that newer versions will follow poolTypeVersion 2 and use tokenRates?

Reply:
Yeah, that probably does make more sense although its worth double checking if any newer versions do come out. Changed.